test/python/circuit/test_controlled_gate.py

```diff
@@ -1002,10 +1015,21 @@ def test_all_inverses(self, gate, num_ctrl_qubits, ctrl_state):
                 numargs = len(_get_free_params(gate))
                 args = [2] * numargs
                 gate = gate(*args)
-                self.assertEqual(
-                    gate.inverse().control(num_ctrl_qubits, ctrl_state=ctrl_state),
-                    gate.control(num_ctrl_qubits, ctrl_state=ctrl_state).inverse(),
+
+                gate_inverse_control = gate.inverse().control(
+                    num_ctrl_qubits, ctrl_state=ctrl_state
+                )
+                gate_control_inverse = gate.control(
+                    num_ctrl_qubits, ctrl_state=ctrl_state
+                ).inverse()
+
+                self.assertTrue(
```

Comment:
What was the reason for changing this from gate equality to matrix equality?

Reply:
At some point the circuits were not equivalent. However, this is now solved and I reverted these changes to test files.